### PR TITLE
Story/ecp 833 ecp error states

### PR DESCRIPTION
**Pull request by martikat**
Updating error message to reflect document

### CODE REVIEW

File: spec/features/admin_claim_with_missing_payroll_gender_spec.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
     click_on "Save and continue"
 
-    expect(page).to have_content("You must select a gender that will be passed to HMRC")
+    expect(page).to have_content("Select male, female, or I don’t know")
```
Comment (WorkSutton) [inline spec/features/admin_claim_with_missing_payroll_gender_spec.rb:25]: Please see the note above relating to payroll gender

File: spec/requests/admin_payroll_gender_tasks_spec.rb [hunk 2]
```diff
@@ -41,7 +41,7 @@
                 post admin_claim_payroll_gender_tasks_path(claim, params: params)
               }.not_to change { claim.tasks.count }
 
-              expect(response.body).to match("You must select a gender that will be passed to HMRC")
+              expect(response.body).to match("Select male, female, or I don’t know")
```
Comment (WorkSutton) [inline spec/requests/admin_payroll_gender_tasks_spec.rb:44]: Please see the notes above relating to payroll gender

**Review by WorkSutton — CHANGES_REQUESTED** [review]
@martikat generally good. A few minor tweaks. Please check with Brian that my assertion about using the latest relevant Error codes was correct. 

As you have correctly identified some of the latest changes are not relevant without changing the pages content, which we have agreed we are not altering for this delivery.

File: app/models/claim.rb [hunk 3]
```diff
@@ -106,15 +106,15 @@ class Claim < ApplicationRecord
 
   validates :academic_year_before_type_cast, format: {with: PolicyConfiguration::ACADEMIC_YEAR_REGEXP}
 
-  validates :payroll_gender, on: [:gender, :submit], presence: {message: "Choose the option for the gender your school’s payroll system associates with you"}
+  validates :payroll_gender, on: [:gender, :submit], presence: {message: "Select male, female, or I don’t know"}
```
[the remark this answers is not in the document]
Comment (martikat) [reply app/models/claim.rb]: I have updated this to v10 now

File: app/models/claim.rb [hunk 4]
```diff
@@ -125,31 +125,31 @@ class Claim < ApplicationRecord
...
                             length: {maximum: 256, message: "Email address must be 256 characters or less"},
                             allow_blank: true
 
-  validates :banking_name, on: [:"bank-details", :submit], presence: {message: "Enter the name on your bank account"}
+  validates :banking_name, on: [:"bank-details", :submit], presence: {message: "Enter a name on the account"}
   validates :bank_sort_code, on: [:"bank-details", :submit], presence: {message: "Enter a sort code"}
   validates :bank_account_number, on: [:"bank-details", :submit], presence: {message: "Enter an account number"}
 
-  validates :payroll_gender, on: [:"payroll-gender-task", :submit], presence: {message: "You must select a gender that will be passed to HMRC"}
+  validates :payroll_gender, on: [:"payroll-gender-task", :submit], presence: {message: "Select male, female, or I don’t know"}
```
[the remark this answers is not in the document]
Comment (martikat) [reply app/models/claim.rb]: I have updated this to v10 now

[the remark this answers is not in the document]
Comment (martikat) [reply app/models/claim.rb]: I have updated this to v10 now

File: spec/features/admin_claim_with_missing_payroll_gender_spec.rb (hunk 1 above)
Comment (martikat) [reply spec/features/admin_claim_with_missing_payroll_gender_spec.rb]: I have updated this to v10 now

File: spec/requests/admin_payroll_gender_tasks_spec.rb (hunk 2 above)
Comment (martikat) [reply spec/requests/admin_payroll_gender_tasks_spec.rb]: I have updated this to v10 now

**Comment by martikat** [comment]
These changes have been covered by other JIRA tickets that cover validation/the error states

**Pull request closed without merging**